domain/src/test/java/org/oppia/android/domain/exploration/ExplorationProgressControllerTest.kt

```diff
@@ -3009,6 +3083,45 @@ class ExplorationProgressControllerTest {
     assertThat(currentState.isHintRevealed(0)).isTrue()
```

Comment:
added an assert statement to ensure the solution is not revealed. 
Also, checking that the index type is `last revealed hint index`  instead of verifying that it is not `show solution` also ensures that the index type case is not some other value (for example index type not set)